analyzers/tests/SonarAnalyzer.UnitTest/TestCases/Hotspots/PermissiveCors.Net.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Net.Http.Headers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+
+namespace Tests.Diagnostics
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CorsEnabledManualAddedHeadersController : Controller
+    {
+        private const string Star = "*";
+
+        [HttpGet]
+        public void Index(string header, string headerValue)
+        {
+            // The Access-Control-Allow-Origin response header indicates whether the response can be shared with requesting code from the given origin.
+            Response.Headers.Add("Access-Control-Allow-Origin", "*"); // Noncompliant
```

Comment:
how about a combination of header names? e.g. from [SO answer](https://stackoverflow.com/a/55016585/9908638)
```
context.Response.Headers.Add(
  CorsConstants.AccessControlAllowHeaders,
  $"{HeaderNames.Authorization}, {HeaderNames.ContentType}, {HeaderNames.AcceptLanguage}, {HeaderNames.Accept}");
```

Reply:
I think the linked example is not correct.

As far as I can tell `context.Response.Headers` is `IDictionary<string, StringValues>` which expects the format "HEADER", "VALUES" (e.g. `context.Response.Headers.Add("Access-Control-Allow-Origin", "*");`).

We could add a test case with concatenated values instead but I think this is not used a lot since the concatenation happens by default due to `StringValues`.